airflow/jobs.py

```diff
@@ -398,9 +398,9 @@ def schedule_dag(self, dag, session=None):
             for dr in active_runs:
                 if (
                         dr.start_date and dag.dagrun_timeout and
-                        dr.start_date < datetime.now() - dag.dagrun_timeout):
+                        dr.start_date < datetime.utcnow() - dag.dagrun_timeout):
```

Comment:
You are assuming here that start_date is in UTC, that is not necessarily the case or even probably not if you dont run in UTC